src/library_dylink.js

```diff
@@ -301,19 +301,33 @@ var LibraryDylink = {
   // that originate from side modules since these are not known at JS
   // generation time.
   $createInvokeFunction__internal: true,
-  $createInvokeFunction__deps: ['$dynCall', 'setThrew'],
+  $createInvokeFunction__deps: ['$dynCall'
+#if !DISABLE_EXCEPTION_CATCHING || SUPPORT_LONGJMP == 'emscripten'
+    , 'setThrew'
+#endif
```

Comment:
Is this optional dependency on setThrew part of that same change?  Or could be be landed on its own maybe?

If we don't call setThrew then can we also just skip the entire `try/catch`?   Which also begs that question, should even include the `createInvokeFunction` function at all in this case?

Reply:
I just opened PR #18665 for this. I think this whole `try/catch` is still needed to restore the stack position whenever an JS exception is raised.

Reply:
If a JS exception is thrown, but we don't handle it.. e.g. because exception handling is not supported, i think the program ends up in an undefined state and there would be no point in restoring the stack pointer.

What do you think about `createInvokeFunction` itself?   Is that only ever needed when longjump and/or exceptions are enabled?

Reply:
> If a JS exception is thrown, but we don't handle it.. e.g. because exception handling is not supported, i think the program ends up in an undefined state and there would be no point in restoring the stack pointer.

This may be true for programs, but for library use (e.g. Embind, which relies on these `invoke_` functions) users may throw a JS exception and catch it on the call site. For example:
https://github.com/GoogleChromeLabs/web-gphoto2/blob/485593b8fed397e5e4e903c6b43bfdbce78ce0d7/api.cpp#L56-L69

IIUC, restoring the stack pointer is then still required, otherwise it could cause undefined behavior.

> What do you think about `createInvokeFunction` itself? Is that only ever needed when longjump and/or exceptions are enabled?

AFAIK, the `createInvokeFunction` function is just a small wrapper around `dynCall` with additional logic for saving and restoring the stack pointer. I'm not sure if it's possible to simplify that logic when Emscripten EH support is not required (e.g. when building with `-fwasm-exceptions -sSUPPORT_LONGJMP=wasm` or `-sSUPPORT_LONGJMP=0`).